R/format_PFN.R

```diff
@@ -190,60 +371,50 @@ format_PFN <- function(db = choose_directory(),
-                      Primary_data$Young5, Primary_data$Young6, Primary_data$Young7,
-                      Primary_data$Young8, Primary_data$Young9, Primary_data$Young10, Primary_data$Young11))
-  #Return only those that don't match the expected ringing format (i.e. XXX9999)
-  #"^[A-Z]{1,}[0-9]{1,}$" is a regular expression that looks for IDs that follow a pattern:
-  #- Starts with at least one capital letter: '^[A-Z]{1,}'
-  #- Ends with at least one number: '[0-9]{1,}$'
-  badIDs <- unique(allIDs[!stringr::str_detect(allIDs, pattern = "^[A-Z]{1,}[0-9]{1,}$")])
-
-
-  ## 1) Rename columns that are equivalent (content and format) to columns in the standard format
-  Brood_data <- Primary_data %>%
+  # 1) Rename columns that are equivalent (content and format) to columns in the standard format
+  Brood_data <- Nest_data %>%
     dplyr::rename(Plot = .data$Popn,
                   LocationID = .data$Box) %>%
 
-    ## 2) Add population ID and reformat colums with equivalent content but different format
-    dplyr::mutate(PopID = "EDM",
+    # 2) Add unique BroodID and reformat columns with equivalent content but different format
+    dplyr::mutate(BroodID = paste0(.data$PopID, "-", .data$BroodID),
                   BreedingSeason = as.integer(Year),
-                  Species = dplyr::case_when(.data$Species == "BLUTI" ~ species_codes[species_codes$SpeciesID == 14620, ]$Species,
+                  Species = dplyr::case_when(.data$Species %in% c("BLUTI", "BLUTI ") ~ species_codes[species_codes$SpeciesID == 14620, ]$Species,
                                              .data$Species == "PIEFL" ~ species_codes[species_codes$SpeciesID == 13490, ]$Species,
-                                             .data$Species == "GRETI" ~ species_codes[species_codes$SpeciesID == 14640, ]$Species,
+                                             .data$Species %in% c("GRETI", "GRETI ") ~ species_codes[species_codes$SpeciesID == 14640, ]$Species,
                                              .data$Species == "REDST" ~ species_codes[species_codes$SpeciesID == 11220, ]$Species,
                                              .data$Species == "MARTI" ~ species_codes[species_codes$SpeciesID == 14400, ]$Species,
                                              .data$Species == "NUTHA" ~ species_codes[species_codes$SpeciesID == 14790, ]$Species,
                                              .data$Species == "COATI" ~ species_codes[species_codes$SpeciesID == 14610, ]$Species,
-                                             .data$Species == "WREN" ~ NA_character_, # Missing, 1 observation only
-                                             .data$Species == "TREEC" ~ NA_character_), # Missing, 1 observation only
+                                             .data$Species == "WREN" ~ NA_character_, # Not currently included, 1 observation only
+                                             .data$Species == "TREEC" ~ NA_character_), # Not currently, 1 observation only
```

Comment:
If you think these could be valid later might be worth adding to `species_codes`

Reply:
I don't think so. These are the "richest" datasets PiedFlyNet administers (as I understand) and there's really not many records here...